### DISCUSSION TITLE
How use the 'Write File' node?

### DISCUSSION THREAD

**Discussion by akordowski** [FlowiseAI/Flowise#2605] [Q&A]
As I could not find any documentation, can any one provide an example how to use the 'Write File' node to write a file? How do I pass content to the file? Thank you in advance.

**Comment by dev-greene** [comment]
I would also love more info.
I've gotten it to work, in a very inconsistent fashion, by asking by bots to save all files to disk when writing code.
Doesn't always work thoguh.

**Comment by brucewedding** [comment]
One more that would like to know how to use the Write File and Read File tools. It's pretty concerning that the documentation has been blank for a year on such a simple and commonly used feature.

**Comment by MaskedLlama04** [comment]
Hey @HenryHengZJ any idea how? Looks like we all are interested. And from what I see, the documents are not finished and I have not found a single tutorial on the internet either

**Comment by MaskedLlama04** [comment]
Guys, nevermind! I found the way. On the Write / Read node, you must tell it both things: 
<img width="525" height="523" alt="image" src="https://github.com/user-attachments/assets/4a8507e1-cb63-4b87-bf26-51ed39fa13ac" />
With the text being whatever you want, and the file_path the name of the file. The directory is wrote on the Write File Parameters.

**Comment by acushlakoncept** [comment]
@akordowski did you make it work?

I added this to the end of my system prompt
```sh
Use the Write File tool to save the response in .md file, with a suitable filename
```
.md is probably just the file type I needed, you can always use whatever file type.

Then add the tool with the `Base_path` of your choice

<img width="673" height="460" alt="Screenshot 2025-09-02 at 06 49 21" src="https://github.com/user-attachments/assets/e15dbe41-46f0-4efa-a331-2db20369a579" />

However I ran into error initially, because it was trying to save it to `/Users/uduakessien/Desktop/desktop`, so I created `desktop` under my `Desktop`.... 

I further updated the `Write File Tool`, though, which allowed it to save to the exact `base_path` I added.
```ts
// packages/components/nodes/tools/WriteFile:82

import * as path from 'path'

 async _call({ file_path, text }: z.infer<typeof this.schema>) {
   const nodeStore = this.store as NodeFileStore

   // Extract just the filename from the path and clean it
   let cleanFileName = path.basename(file_path)

   await this.store.writeFile(cleanFileName, text)
   return 'File written to successfully updated.'
 }
```

**Reply by akordowski** [reply]
No, this was a onetime test long time ago 😉
